mkosi

```diff
@@ -2011,6 +2026,10 @@ def unlink_output(args):
             with complete_step('Clearing out build directory'):
                 empty_directory(args.build_dir)
 
+        if args.cache_path is not None:
+            with complete_step('Clearing out package cache'):
+                empty_directory(args.cache_path)
```

Comment:
I'm not so sure about this one. Unlike the build dir and the mkosi cache, packages are uniquely named: either you want file foo_something.deb or you don't: the contents of foo_something.deb cannot change (and if they do, the relevant apt/rpm command would reject the invalid signature). At least in the deb world, that holds true, and I think it holds true in the rpm world too. I have no idea about arch.

Therefore, I think cleaning the package cache should not be mixed with clearing mkosi stuff or build stuff.

Reply:
Arch package file names also contain the version and should therefore be unique as well. On the other hand, hiding this behind two/three `--force`s seems acceptable to me.